src/clients/HubPoolClient.ts

```diff
@@ -771,17 +771,37 @@ export class HubPoolClient extends BaseAbstractClient {
       return { success: false };
     }
 
+    const eventSearchConfigs = eventNames.map((eventName) => {
+      if (!Object.keys(hubPoolEvents).includes(eventName)) {
+        throw new Error(`HubPoolClient: Cannot query unrecognised HubPool event name: ${eventName}`);
+      }
+
+      const _searchConfig = { ...searchConfig }; // shallow copy
+
+      // By default, an event's query range is controlled by the `searchConfig` passed in during
+      // instantiation. However, certain events generally must be queried back to HubPool genesis.
+      const overrideEvents = ["CrossChainContractsSet", "L1TokenEnabledForLiquidityProvision", "SetPoolRebalanceRoute"];
+      if (overrideEvents.includes(eventName) && !this.isUpdated) {
+        _searchConfig.fromBlock = this.deploymentBlock;
+      }
+
+      return {
+        eventName,
+        filter: hubPoolEvents[eventName],
+        searchConfig: _searchConfig,
+      };
+    });
+
     this.logger.debug({
       at: "HubPoolClient",
       message: "Updating HubPool client",
-      searchConfig,
-      eventNames,
+      searchConfig: eventSearchConfigs.map(({ eventName, searchConfig }) => ({ eventName, searchConfig })),
```

Comment:
Here's an example of this log message to show the overriding working:

```
2024-03-09 10:24:00 [debug]: {                                                                                                                                                                                                   
  "at": "HubPoolClient",                                                                                                                                                                                                         
  "message": "Updating HubPool client",                                                                                                                                                                                          
  "searchConfig": [  
    {                 
      "eventName": "SetPoolRebalanceRoute",
      "searchConfig": { 
        "fromBlock": 14819537,
        "toBlock": 19396921,
        "maxBlockLookBack": 10000
      }                           
    },                                 
    {                         
      "eventName": "L1TokenEnabledForLiquidityProvision",                                                       
      "searchConfig": {
        "fromBlock": 14819537,
        "toBlock": 19396921,     
        "maxBlockLookBack": 10000
      }            
    },          
    {             
      "eventName": "ProposeRootBundle",
      "searchConfig": {
        "fromBlock": 19393465,
        "toBlock": 19396921,
        "maxBlockLookBack": 10000
      }            
    },           
    {            
      "eventName": "RootBundleCanceled",
      "searchConfig": {
        "fromBlock": 19393465,
        "toBlock": 19396921,
        "maxBlockLookBack": 10000
      }             
    },                  
    {                     
      "eventName": "RootBundleDisputed",
      "searchConfig": {        
        "fromBlock": 19393465,
        "toBlock": 19396921, 
        "maxBlockLookBack": 10000
      }                               
    },                                  
    {                                 
      "eventName": "RootBundleExecuted",
      "searchConfig": {                                                                                         
        "fromBlock": 19393465,     
        "toBlock": 19396921,                 
        "maxBlockLookBack": 10000                
      }                                                                                                         
    },                                       
    {                                      
      "eventName": "CrossChainContractsSet",
      "searchConfig": {         
        "fromBlock": 14819537,
        "toBlock": 19396921, 
        "maxBlockLookBack": 10000
      }                            
    }             
  ],                          
}                                          
2024-03-09 10:24:02 [debug]: {     
  "at": "HubPoolClient#_update",
  "message": "Time to query new events from RPC for 1: 2311 ms",
}
```

Reply:
2300 ms is a huge speed up from 28000 ms --> 10x!